File: app/src/main/java/de/christinecoenen/code/zapp/app/settings/helper/PreferenceHelper.java
package de.christinecoenen.code.zapp.app.settings.helper;


import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;
import java.util.List;

class PreferenceHelper {

	private static final String SHARED_PEFERENCES_NAME = "ZAPP_SHARED_PREFERENCES";

	private final Gson gson = new Gson();
	private final SharedPreferences preferences;

	PreferenceHelper(Context context) {
		preferences = context.getSharedPreferences(SHARED_PEFERENCES_NAME, Context.MODE_PRIVATE);
	}

	void saveList(String key, List list) {
		SharedPreferences.Editor editor = preferences.edit();
		String jsonList = gson.toJson(list);
		editor.putString(key, jsonList);
		editor.apply();
	}

	<T> List<T> loadList(String key) {
		if (preferences.contains(key)) {
			String jsonList = preferences.getString(key, null);
			Type listType = new TypeToken<ArrayList<T>>(){}.getType();
			return gson.fromJson(jsonList, listType);
		} else {
			return null;
		}
	}
}

File: app/src/main/java/de/christinecoenen/code/zapp/app/settings/helper/PreferenceHelper.kt
package de.christinecoenen.code.zapp.app.settings.helper

import android.content.Context
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import java.util.*

internal class PreferenceHelper(context: Context) {

	companion object {
		private const val SHARED_PEFERENCES_NAME = "ZAPP_SHARED_PREFERENCES"
	}


	private val gson = Gson()
	private val preferences = context.getSharedPreferences(SHARED_PEFERENCES_NAME, Context.MODE_PRIVATE)


	fun saveList(key: String, list: List<String>) {
		val jsonList = gson.toJson(list)

		preferences.edit()
			.putString(key, jsonList)
			.apply()
	}

	fun loadList(key: String): List<String>? {
		if (!preferences.contains(key)) {
			return null
		}

		val jsonList = preferences.getString(key, null)
		val listType = object : TypeToken<ArrayList<String>?>() {}.type

		return gson.fromJson(jsonList, listType)
	}
}
